### ISSUE TITLE
 Error: Field X is not present in the Query Processor Store.

### ISSUE THREAD

**Issue by AgustinBanchio** [metabase/metabase#34950]
### Describe the bug

In some of our saved questions, when we try to load them we get this error:

` Error: Field X is not present in the Query Processor Store.`

Where the Field number can change depending on the question. 
If editing the question it starts working, even if setting it back to the previous values, once the questions becomes saved, then they break even, either when overwriting or saving into a new one.

This only started to happen recently, possibly after the 0.47.5 update.

### To Reproduce

Unfortunately, I don't know what caused the issue with the faulty saved questions initially.

### Expected behavior

Saved questions should load fine in dashboard or by themselves.

### Logs

```
[443016d5-a474-4579-bbce-e10a8b00543b] 2023-10-23T10:36:22+01:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: Error: Field 140 is not present in the Query Processor Store.
{:database_id 2,
 :started_at #t "2023-10-23T09:36:22.358526Z[GMT]",
 :json_query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :query,
  :middleware {:js-int-to-string? true, :ignore-cached-results? false, :process-viz-settings? false},
  :database 2,
  :query
  {:source-table 127,
   :joins
   [{:fields :all,
     :alias "Organisations - Client",
     :condition [:= [:field 1255 nil] [:field 692 {:join-alias "Organisations - Client"}]],
     :source-table 75}],
   :aggregation [[:count]],
   :breakout [[:field 1270 {:temporal-unit :month}]],
   :filter
   [:and
    [:> [:field 1270 nil] "2021-01-01"]
    [:!= [:field 706 {:join-alias "Organisations - Client"}] "Risk Ledger (Supplier Demo)"]]},
  :parameters [],
  :async? true,
  :cache-ttl 13},
 :native nil,
 :status :failed,
 :class java.lang.Exception,
 :stacktrace
 ["--> query_processor.store$default_field.invokeStatic(store.clj:270)"
  "query_processor.store$default_field.invoke(store.clj:266)"
  "query_processor.store$fn__60333$field__60338$fn__60339.invoke(store.clj:279)"
  "query_processor.store$fn__60333$field__60338.invoke(store.clj:276)"
  "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing.invokeStatic(validate_temporal_bucketing.clj:36)"
  "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing.invoke(validate_temporal_bucketing.clj:30)"
  "query_processor$preprocess_STAR_$fn__78297.invoke(query_processor.clj:167)"
  "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:165)"
  "query_processor$preprocess_STAR_.invoke(query_processor.clj:160)"
  "query_processor$fn__78305$combined_pre_process__78306$combined_pre_process_STAR___78307.invoke(query_processor.clj:257)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__76983$fn__76988.invoke(resolve_database_and_driver.clj:36)"
  "driver$do_with_driver.invokeStatic(driver.clj:93)"
  "driver$do_with_driver.invoke(driver.clj:88)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__76983.invoke(resolve_database_and_driver.clj:35)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__72965.invoke(fetch_source_query.clj:316)"
  "query_processor.middleware.store$initialize_store$fn__73146$fn__73147.invoke(store.clj:12)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:56)"
  "query_processor.store$do_with_store.invoke(store.clj:50)"
  "query_processor.middleware.store$initialize_store$fn__73146.invoke(store.clj:11)"
  "query_processor.middleware.normalize_query$normalize$fn__77279.invoke(normalize_query.clj:36)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__74673.invoke(constraints.clj:54)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__77208.invoke(process_userland_query.clj:151)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__77605.invoke(catch_exceptions.clj:171)"
  "query_processor.reducible$async_qp$qp_STAR___66814$thunk__66816.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___66814$fn__66818.invoke(reducible.clj:108)"],
 :card_id 229,
 :context :question,
 :error "Error: Field 140 is not present in the Query Processor Store.",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :data {:rows [], :cols []}}
```

### Information about your Metabase installation

```JSON
Our DB is BigQuery
Latest Metabase version 0.47.5
The internal metabase DB is Postgres.
Metabase is docker hosted on AWS ECS.
```

### Severity

blocking some users

**Comment by paoliniluis** [comment]
Have you done any change to the DB?

**Comment by AgustinBanchio** [comment]
> Have you done any change to the DB?

No, there have not been any changes to the DB. It's hosted in AWS as a Postgres 14.7 RDS instance.
Do you think it could have had an issue when upgrading Metabase to the new version?

There are many questions that still work fine.

Is it possible to downgrade to Metabase 0.47.4 and see if the error persists? Or could it cause more trouble?

**Comment by jeremford** [comment]
I am also now seeing the same exception on 0.47.5.  There were no issues of this type for 0.47.4.

I'm using AWS Aurora Postgres 15.3 (for both Metabase itself, and the database I'm reporting on).

I get this issue on Questions created on the new version, and also existing questions.

It appears that the Question works fine, but after adding it to a Dashboard it is unable to render, and shows that error.

**Comment by MarcSamD** [comment]
My 2 cents:
- I confirm that it is a new bug introduced by 0.47.5
- I have plenty of questions and dashboard but this issue only happens for 2 questions, and I'm not able to identify what could be unique with them.
- Removing a filter on the question make it works. Adding exactly the same filter will also still work. But as soon as we save it (with or without the filters), it cannot be render again.

**Comment by sztanko** [comment]
We also see this bug since 0.47.5 
Postgres RDS.

**Comment by paoliniluis** [comment]
@MarcSamD can you provide us logs? we would need backend logs when you do the save and also the frontend logs when the question works and when the question breaks. If possible, send us the exact payloads of the API's

**Comment by paoliniluis** [comment]
For everyone that's hitting this error, moving back to 47.4 will fix the issue, but we need to come up with a consistent reproduction of this. 

Can someone give us any hint about:
1) is is a SQL or a GUI question?
2) does it happen randomly?

**Comment by paoliniluis** [comment]
EDIT: was the question created with an older version of Metabase?

**Comment by sztanko** [comment]
1. It is a GUI question in our case
2. It happens consistently on the same questions, but only small quotient of questions is affected. If I modify the question (e.g. add a new filter, then remove it), it would work, but once I save it, it stops working again. Permalink to unsaved questions works and doesn't five an error.
3. Yes, dashboards where created with old version of Metabase. However, modifying the old question and saving it under new name would still give us an error.
4. It does run on Postgres.

**Comment by paoliniluis** [comment]
@sztanko can you give me an example of the question? I would like to get something similar. I also need the structure of the table if possible

**Comment by sicarul** [comment]
I'm having the same issue since 0.47.5 on a Snowflake database (MySQL Backend)

**Comment by paoliniluis** [comment]
@sicarul can you please give us some insights about the question that fails? is that question a simple question? a question of other question? does it have any custom column? anything that can give us insights to debug?

**Comment by paoliniluis** [comment]
@sicarul so creating the questions again fixed those?

**Comment by sicarul** [comment]
@paoliniluis what i'm saying is that there was no way to fix them, so i had to archive the existing questions and create them from scratch, to add the new, working questions to the dashboards.

**Comment by paoliniluis** [comment]
@sicarul exactly, so you needed to recreate everything. And the new questions worked without issues right?

**Comment by sicarul** [comment]
@paoliniluis exactly!

**Comment by luizarakaki** [comment]
@sicarul are these questions pivot tables?

**Comment by sicarul** [comment]
@paoliniluis no, all i found failing so far were regular tables.

**Comment by AgustinBanchio** [comment]
> @AgustinBanchio @jeremford @MarcSamD @sztanko what database does the question run on? we're looking for the DW, not the application database

Our data warehouse is on BigQuery

**Comment by paoliniluis** [comment]
@AgustinBanchio please find field 140 to see what's the table that belongs to. To find this field you need to go to settings->admin->table metadata and search for the field in the tables

e.g. https://<yourmetabaseurl>/admin/datamodel/database/1/schema/1:PUBLIC/table/1/field/1/general

in the above url that's field with id 1(field/1/general)

can you locate the field and tell us which table id it belongs to? does it belong to table 127 or to table 75?

**Comment by AgustinBanchio** [comment]
> @AgustinBanchio please find field 140 to see what's the table that belongs to. To find this field you need to go to settings->admin->table metadata and search for the field in the tables
> 
> e.g. https:///admin/datamodel/database/1/schema/1:PUBLIC/table/1/field/1/general
> 
> in the above url that's field with id 1(field/1/general)
> 
> can you locate the field and tell us which table id it belongs to? does it belong to table 127 or to table 75?

Apparently, it doesn't belong to any of those tables.
This is what I get with any of the two tables.
<img width="622" alt="image" src="https://github.com/metabase/metabase/assets/19520160/d805993a-071b-45b7-ae61-d537e36b0323">

**Comment by AgustinBanchio** [comment]
I can provide the API response when trying to view one of the failing questions:

```
{
    "database_id": 2,
    "started_at": "2023-10-24T17:15:46.441544Z",
    "json_query": {
        "constraints": {
            "max-results": 10000,
            "max-results-bare-rows": 2000
        },
        "type": "query",
        "middleware": {
            "js-int-to-string?": true,
            "ignore-cached-results?": true,
            "process-viz-settings?": false
        },
        "database": 2,
        "query": {
            "source-table": 127,
            "joins": [
                {
                    "fields": "all",
                    "alias": "Organisations - Client",
                    "condition": [
                        "=",
                        [
                            "field",
                            1255,
                            null
                        ],
                        [
                            "field",
                            692,
                            {
                                "join-alias": "Organisations - Client"
                            }
                        ]
                    ],
                    "source-table": 75
                }
            ],
            "aggregation": [
                [
                    "count"
                ]
            ],
            "breakout": [
                [
                    "field",
                    1270,
                    {
                        "temporal-unit": "quarter"
                    }
                ]
            ],
            "filter": [
                "and",
                [
                    "!=",
                    [
                        "field",
                        706,
                        {
                            "join-alias": "Organisations - Client"
                        }
                    ],
                    "Risk Ledger (Supplier Demo)"
                ],
                [
                    ">",
                    [
                        "field",
                        1270,
                        null
                    ],
                    "2021-01-01"
                ]
            ]
        },
        "parameters": [],
        "async?": true,
        "cache-ttl": 4
    },
    "native": null,
    "status": "failed",
    "class": "class java.lang.Exception",
    "stacktrace": [
        "--> query_processor.store$default_field.invokeStatic(store.clj:270)",
        "query_processor.store$default_field.invoke(store.clj:266)",
        "query_processor.store$fn__60333$field__60338$fn__60339.invoke(store.clj:279)",
        "query_processor.store$fn__60333$field__60338.invoke(store.clj:276)",
        "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing.invokeStatic(validate_temporal_bucketing.clj:36)",
        "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing.invoke(validate_temporal_bucketing.clj:30)",
        "query_processor$preprocess_STAR_$fn__78297.invoke(query_processor.clj:167)",
        "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:165)",
        "query_processor$preprocess_STAR_.invoke(query_processor.clj:160)",
        "query_processor$fn__78305$combined_pre_process__78306$combined_pre_process_STAR___78307.invoke(query_processor.clj:257)",
        "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__76983$fn__76988.invoke(resolve_database_and_driver.clj:36)",
        "driver$do_with_driver.invokeStatic(driver.clj:93)",
        "driver$do_with_driver.invoke(driver.clj:88)",
        "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__76983.invoke(resolve_database_and_driver.clj:35)",
        "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__72965.invoke(fetch_source_query.clj:316)",
        "query_processor.middleware.store$initialize_store$fn__73146$fn__73147.invoke(store.clj:12)",
        "query_processor.store$do_with_store.invokeStatic(store.clj:56)",
        "query_processor.store$do_with_store.invoke(store.clj:50)",
        "query_processor.middleware.store$initialize_store$fn__73146.invoke(store.clj:11)",
        "query_processor.middleware.normalize_query$normalize$fn__77279.invoke(normalize_query.clj:36)",
        "query_processor.middleware.constraints$add_default_userland_constraints$fn__74673.invoke(constraints.clj:54)",
        "query_processor.middleware.process_userland_query$process_userland_query$fn__77208.invoke(process_userland_query.clj:151)",
        "query_processor.middleware.catch_exceptions$catch_exceptions$fn__77605.invoke(catch_exceptions.clj:171)",
        "query_processor.reducible$async_qp$qp_STAR___66814$thunk__66816.invoke(reducible.clj:103)",
        "query_processor.reducible$async_qp$qp_STAR___66814$fn__66818.invoke(reducible.clj:108)"
    ],
    "card_id": 230,
    "context": "question",
    "error": "Error: Field 140 is not present in the Query Processor Store.",
    "row_count": 0,
    "running_time": 0,
    "preprocessed": null,
    "data": {
        "rows": [],
        "cols": []
    }
}
```

**Comment by AgustinBanchio** [comment]
Alright I connected to the Metabase DB manually.

Field 140 is a field called created_at (we have one of those usually on almost all of our tables).
Belongs to table 18 which is an old table that is not active (was deleted from our bigquery warehouse) and was replaced with table 127 a long time ago.

So could the original question have been on that table maybe and it was later updated to the new table. Don't know the internals of metabase but maybe it saves the question history and for some reason it is now trying to use or validate parts of the history that isn't valid anymore?

**Comment by luizarakaki** [comment]
We don't update references like this... Was this field critical for the query? Or could it be ignored to build the viz?

**Comment by AgustinBanchio** [comment]
> We don't update references like this... Was this field critical for the query? Or could it be ignored to build the viz?

The question doesn't even involve that table and field anymore. After the table was removed more than a year ago, the question was updated to use a new table and fields from the new table. This timestamp field with id 140 was used as a filter and also to group by. It should use field 1270 (also called created_at). 

Field 140 or it's table are not referenced in any part of the question

**Comment by luizarakaki** [comment]
I see... it is indeed not part of the query object...

**Comment by AgustinBanchio** [comment]
Could it be trying to match a field by name? The field name is the same and the table name is very similar, one is singular the other is plural

**Comment by luizarakaki** [comment]
We don't do this kind of matching... the field/table name is irrelevant. I guess this is related to implicit joins

**Comment by luizarakaki** [comment]
Can you check what returns from:

```
select id, fk_target_field_id
from metabase_field
where
    table_id = 18
```

**Comment by AgustinBanchio** [comment]
> select id, fk_target_field_id
> from metabase_field
> where
>     table_id = 18

It gives me this:

<img width="363" alt="image" src="https://github.com/metabase/metabase/assets/19520160/9d039389-6e57-45f9-ad95-7327cebdb56b">

**Comment by luizarakaki** [comment]
Can you now check which tables these fields (398 and 692) are part of?

**Comment by AgustinBanchio** [comment]
Field 398 is from table 28
And field 692 is from table 75

**Comment by luizarakaki** [comment]
Got it. Thank you very much. I think the path to repro is that table 18 is inactive but implicit joined to table 75 (present in the query).

Let me try some experiments here

**Comment by AgustinBanchio** [comment]
> Got it. Thank you very much. I think the path to repro is that table 18 is inactive but implicit joined to table 75 (present in the query).
> 
> Let me try some experiments here

Could it be related to this change?
https://github.com/metabase/metabase/commit/d4af9abcff738b45a42a07b8d015341f73c12716

**Comment by luizarakaki** [comment]
Yeah, I'd say this is the best candidate

**Comment by pulse-mind** [comment]
Same behavior for me.

**Comment by cryptopoly** [comment]
Yes also experiencing the same issue with connection to MongoDB. Saved question errors on dashboard and independently. Removing and adding the same filter resolves but when saved again the error returns. No joins in the query, just summarised view with some filters

**Comment by camsaul** [comment]
@AgustinBanchio also 

```sql
SELECT result_metadata FROM report_card WHERE id = 230;
```

**Comment by AgustinBanchio** [comment]
I've downgraded to 0.47.4 as it doesn't have the issue.
Would the data returned from the query still be valid?

**Comment by camsaul** [comment]
One more actually too if you can:

```sql
SELECT visualization_settings FROM report_card WHERE id = 230;
```

**Comment by AgustinBanchio** [comment]
Some of the queries reveal excessive metadata related to our warehouse, including table and field names. I am consulting internally to ensure that sharing specific details relevant to this issue is permissible. While I believe there's minimal risk, I want to confirm that I have the clearance to share this information. I appreciate your understanding and will update you once I receive a response.

**Comment by camsaul** [comment]
@AgustinBanchio In that case, can you just leave the Table/Field names out? I'm mostly interested in figuring out how that hop to Field 140 is getting introduced, we should still be able to work that out with just the IDs.

**Comment by camsaul** [comment]
Actually I think I've figured this out now, 

```sql
SELECT visualization_settings FROM report_card WHERE id = 230;
```

would help confirm my theory but I don't need the other two any more.

**Comment by AgustinBanchio** [comment]
> Actually I think I've figured this out now,
> 
> ```sql
> SELECT visualization_settings FROM report_card WHERE id = 230;
> ```
> 
> would help confirm my theory but I don't need the other two any more.

Would it be ok if I remove the column names from the json and just leave the field refs?

**Comment by paoliniluis** [comment]
@AgustinBanchio please redact them if they’re not many

**Comment by AgustinBanchio** [comment]
Here is the response with the field names removed.

```json
{
  "table.columns": [
    {
      "fieldRef": [
        "field",
        706,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "fieldRef": [
        "field",
        138,
        null
      ],
      "enabled": true
    },
    {
      "fieldRef": [
        "field",
        140,
        {
          "temporal-unit": "default"
        }
      ],
      "enabled": true
    },
    {
      "fieldRef": [
        "field",
        147,
        null
      ],
      "enabled": true
    },
    {
      "fieldRef": [
        "field",
        149,
        {
          "temporal-unit": "default"
        }
      ],
      "enabled": true
    },
    {
      "fieldRef": [
        "field",
        142,
        {
          "temporal-unit": "default"
        }
      ],
      "enabled": true
    },
    {
      "fieldRef": [
        "field",
        143,
        {
          "temporal-unit": "default"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        692,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        687,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        695,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        725,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        726,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        691,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        706,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        703,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        684,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        705,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        699,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        681,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        698,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        685,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        811,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        709,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        702,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        704,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        686,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        696,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        701,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        688,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        682,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        707,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        693,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        700,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        689,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        697,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        694,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        708,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        690,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        1169,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        1168,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        1170,
        {
          "join-alias": "Organisations - Client"
        }
      ],
      "enabled": true
    },
    {
      "field_ref": [
        "field",
        1270,
        null
      ],
      "enabled": true
    }
  ],
  "graph.dimensions": [
    "created_at"
  ],
  "graph.metrics": [
    "count"
  ]
}

```

**Comment by paoliniluis** [comment]
@AgustinBanchio will you be able to test a build with the fix?

**Comment by camsaul** [comment]
yup, looks like my theory is confirmed, the viz settings map has references to fields that aren't in the query anymore, after #34583 landed (in 47.5) this causes errors since it's accidentally trying to validate those Field refs

**Comment by camsaul** [comment]
Fixed by #35071

**Comment by paoliniluis** [comment]
@sztanko @sicarul @AgustinBanchio 47.6 is out, please test and post your findings

**Comment by MarcSamD** [comment]
I confirm that updating to 47.6 fix the issue (no additional steps required, just refresh the page).

**Comment by AgustinBanchio** [comment]
Sorry for the delay, can also confirm 47.6 fixes the issue!
Thanks!

**Comment by lucasmocellin** [comment]
@camsaul I have 47.4 and I am experience this:

> Error: Field 3,762 is not present in the Query Processor Store.

Should I expect this update to 47.6 to fix it as well and consider the same problem or not?

diagnostics info:
```
{
  "browser-info": {
    "language": "en",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.20.1+1",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.20.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.20.1+1",
    "os.name": "Linux",
    "os.version": "4.14.330-176.540.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "US/Pacific"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mysql",
      "athena"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "8.0.33"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.7.6"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-10-11",
      "tag": "v0.47.4",
      "branch": "?",
      "hash": "c96dc65"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

I can provide additional information if that helps.

**Comment by camsaul** [comment]
@lucasmocellin updating to 47.6 should probably fix it (looks like the same issue everyone else was having), but if not try upgrading to 48.x, we eliminated that error entirely in 48.0.
